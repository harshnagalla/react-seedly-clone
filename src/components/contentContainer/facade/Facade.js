import style from "./Facade.module.scss";
import fetch from "isomorphic-unfetch";
import React, { useState, useEffect } from "react";

const Facade = ({ headerText, secondaryHeaderText, buttonText }) => {
  const [state, setstate] = useState();
  useEffect(() => {
    fetch(`http://localhost:7000/topics`)
      .then((response) => response.json())
      .then((data) => {
        setstate(data);
      });
  }, []);

  console.log("Data to fetch", state);
  return (
    <div className={style.facade}>
      <label className={style.facade__header}>{headerText}</label>
      <label className={style.facade__secondaryHeader}>
        {secondaryHeaderText}
      </label>
      <button className={style.facade__button}>{buttonText}</button>
    </div>
  );
};

export default Facade;
