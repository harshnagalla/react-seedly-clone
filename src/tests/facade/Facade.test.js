import React from "react";
import { cleanup, render } from "@testing-library/react";
import Facade from "../../components/contentContainer/facade/Facade";

const getTestComponent = () => {
  return (
    <Facade
      headerText={"Headers"}
      secondaryHeaderText={"This is the answer"}
      buttonText={"Add Question"}
    ></Facade>
  );
};

describe("Snapshot Testing", () => {
  test("Facade snapshot", () => {
    const facadePost = render(
      <Facade
        headerText={"Headers"}
        secondaryHeaderText={"This is the answer"}
        buttonText={"Add Question"}
      ></Facade>
    );
    expect(facadePost.container).toMatchSnapshot();
  });
});

describe("Facade Test", () => {
  it("Should have header", () => {
    const { queryByText } = render(getTestComponent());
    expect(queryByText("Headers")).not.toBeNull();
  });
  it("Should have secondaryHeader", () => {
    const { queryByText } = render(getTestComponent());
    expect(queryByText("This is the answer")).not.toBeNull();
  });
  it("Should have button text", () => {
    const { queryByText } = render(getTestComponent());
    expect(queryByText("Add Question")).not.toBeNull();
  });
});
