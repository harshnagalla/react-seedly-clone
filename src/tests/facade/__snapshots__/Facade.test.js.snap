// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing Facade snapshot 1`] = `
<div>
  <div
    class="facade"
  >
    <label
      class="facade__header"
    >
      Headers
    </label>
    <label
      class="facade__secondaryHeader"
    >
      This is the answer
    </label>
    <button
      class="facade__button"
    >
      Add Question
    </button>
  </div>
</div>
`;
